### PR TITLE
#218 Allow user to hide sidebar

### PR DESCRIPTION
**Pull request by pbson**
Partially fix: https://github.com/OpenPaaS-Suite/esn-frontend-calendar/issues/218
Demo: 
sideee

### CODE REVIEW

**Review by rezk2ll — APPROVED** [review]

File: src/linagora.esn.calendar/app/components/show-main-sidebar-button/show-main-sidebar-button.component.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+(function(angular) {
```
Comment (huy-ta) [inline src/linagora.esn.calendar/app/components/show-main-sidebar-button/show-main-sidebar-button.component.js:1]: No need for IIFEs.

File: src/linagora.esn.calendar/app/components/show-main-sidebar-button/show-main-sidebar-button.controller.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+(function(angular) {
+  'use strict';
+
+  angular.module('esn.calendar')
+    .run(function($rootScope) {
+      $rootScope.hideSidebar = true;
```
Comment (huy-ta) [inline src/linagora.esn.calendar/app/components/show-main-sidebar-button/show-main-sidebar-button.controller.js:6]: No need for IIFEs.

File: src/linagora.esn.calendar/app/components/show-main-sidebar-button/show-main-sidebar-button.controller.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    })
+    .controller('CalShowMainSidebarButtonController', CalShowMainSidebarButtonController);
+
+  function CalShowMainSidebarButtonController($rootScope) {
+    var self = this;
```
Comment (huy-ta) [inline src/linagora.esn.calendar/app/components/show-main-sidebar-button/show-main-sidebar-button.controller.js:11]:
```suggestion
    const self = this;
```

File: src/linagora.esn.calendar/app/components/show-main-sidebar-button/show-main-sidebar-button.pug [hunk 4]
```diff
@@ -0,0 +1,2 @@
+button.btn.btn-link.btn-icon.waves-notransition.hidden-xs(ng-click="ctrl.toggleDisplay()", title="{{ 'Show planning' | translate }}")
```
Comment (huy-ta) [inline src/linagora.esn.calendar/app/components/show-main-sidebar-button/show-main-sidebar-button.pug:1]: `Show planning`? :trollface:

File: src/linagora.esn.calendar/app/components/show-main-sidebar-button/show-main-sidebar-button.controller.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+angular.module('esn.calendar')
+  .run(function($rootScope) {
+    $rootScope.hideSidebar = true;
+  })
+  .controller('CalShowMainSidebarButtonController', CalShowMainSidebarButtonController);
+function CalShowMainSidebarButtonController($rootScope) {
+  const self = this;
+
+  self.toggleDisplay = toggleDisplay;
```
Comment (rezk2ll) [inline src/linagora.esn.calendar/app/components/show-main-sidebar-button/show-main-sidebar-button.controller.js:11]:
```suggestion
  self.toggleDisplay = function {
    $rootScope.hideSidebar = !$rootScope.hideSidebar;
  }
```

File: src/linagora.esn.calendar/app/index.pug [hunk 6]
```diff
@@ -2,7 +2,7 @@
   .col-xs-12(ng-if="alertMessage")
     span.calendar-error-message {{alertMessage}}
   .calendar-view
-    .col-md-3.col-lg-2.hidden-xs.hidden-sm
+    .col-md-3.col-lg-2.hidden-xs.hidden-sm(ng-if='hideSidebar')
```
Comment (rezk2ll) [inline src/linagora.esn.calendar/app/index.pug:5]: so this here means show the `cal-sidebar` component when the `hideSidebar` is true.

I suggest renaming `hideSidebar` to `showSidebar` cause it makes more sense when reading. or just invert it to `ng-if='!hideSidebar'` and invert the initial value in your run function:

```js
.run(function($rootScope) {
  $rootScope.hideSidebar = false; // Shown by default
})

```

File: src/esn.calendar.libs/app/services/sidebar-visibility.service.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  let isSidebarVisible = true;
+
+  function getSidebarVisibility() {
+    return isSidebarVisible;
+  }
+
+  function setSidebarVisibility(sidebarVisibility) {
+    isSidebarVisible = sidebarVisibility;
+
+    $rootScope.$broadcast('change-sidebar-visibility');
```
Comment (rezk2ll) [inline src/esn.calendar.libs/app/services/sidebar-visibility.service.js:14]: just a small comment, the `change-sidebar-visibility` should be declared in an angular constant.

File: src/linagora.esn.calendar/app/calendar-root/calendar-root.controller.js [hunk 8]
```diff
@@ -4,11 +4,16 @@
...
 
-  function CalCalendarRootController($scope, calendarService, calDefaultValue, calendarHomeId, businessHours) {
+  function CalCalendarRootController($rootScope, $scope, calendarService, calDefaultValue, calendarHomeId, businessHours, calendarSidebarService) {
     var CAL_DEFAULT_OLD_CALENDAR_ID = 'events';
 
     $scope.calendarHomeId = calendarHomeId;
     $scope.businessHours = businessHours;
+    $scope.isSidebarVisible = calendarSidebarService.getSidebarVisibility();
+
+    $rootScope.$on('change-sidebar-visibility', function() {
```
Comment (rezk2ll) [inline src/linagora.esn.calendar/app/calendar-root/calendar-root.controller.js:14]: same here.

**Review by rezk2ll — DISMISSED** [review]

File: src/esn.calendar.libs/app/services/sidebar-visibility.service.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
+angular.module('esn.calendar.libs')
+  .service('calendarSidebarService', calendarSidebarService);
+
+function calendarSidebarService($rootScope) {
```
Comment (rezk2ll) [inline src/esn.calendar.libs/app/services/sidebar-visibility.service.js:4]:
```suggestion
function calendarSidebarService($rootScope, CAL_SIDEBAR_VISIBILITY) {
```

File: src/esn.calendar.libs/app/services/sidebar-visibility.service.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+  let isSidebarVisible = true;
+
+  function getSidebarVisibility() {
+    return isSidebarVisible;
+  }
+
+  function setSidebarVisibility(sidebarVisibility) {
+    isSidebarVisible = sidebarVisibility;
+
+    $rootScope.$broadcast('CAL_SIDEBAR_VISIBILITY');
```
Comment (rezk2ll) [inline src/esn.calendar.libs/app/services/sidebar-visibility.service.js:14]:
```suggestion
    $rootScope.$broadcast(CAL_SIDEBAR_VISIBILITY);
```

File: src/linagora.esn.calendar/app/calendar-root/calendar-root.controller.js [hunk 11]
```diff
@@ -4,11 +4,16 @@
   angular.module('esn.calendar')
     .controller('CalCalendarRootController', CalCalendarRootController);
 
-  function CalCalendarRootController($scope, calendarService, calDefaultValue, calendarHomeId, businessHours) {
+  function CalCalendarRootController($rootScope, $scope, calendarService, calDefaultValue, calendarHomeId, businessHours, calendarSidebarService) {
```
Comment (rezk2ll) [inline src/linagora.esn.calendar/app/calendar-root/calendar-root.controller.js:7]:
```suggestion
  function CalCalendarRootController($rootScope, $scope, calendarService, calDefaultValue, calendarHomeId, businessHours, calendarSidebarService, CAL_SIDEBAR_VISIBILITY) {
```

File: src/linagora.esn.calendar/app/calendar-root/calendar-root.controller.js [hunk 12]
```diff
@@ -4,11 +4,16 @@
...
     var CAL_DEFAULT_OLD_CALENDAR_ID = 'events';
 
     $scope.calendarHomeId = calendarHomeId;
     $scope.businessHours = businessHours;
+    $scope.isSidebarVisible = calendarSidebarService.getSidebarVisibility();
+
+    $rootScope.$on('CAL_SIDEBAR_VISIBILITY', function() {
```
Comment (rezk2ll) [inline src/linagora.esn.calendar/app/calendar-root/calendar-root.controller.js:14]:
```suggestion
    $rootScope.$on(CAL_SIDEBAR_VISIBILITY, function() {
```

**Review by rezk2ll — APPROVED** [review]

**Review by huy-ta — APPROVED** [review]

**Pull request merged**